cluster-autoscaler/cloudprovider/azure/azure_config.go

```diff
@@ -81,70 +64,49 @@ type CloudProviderRateLimitConfig struct {
 
 // Config holds the configuration parsed from the --cloud-config flag
 type Config struct {
-	CloudProviderRateLimitConfig
-
-	Cloud          string `json:"cloud" yaml:"cloud"`
-	Location       string `json:"location" yaml:"location"`
-	TenantID       string `json:"tenantId" yaml:"tenantId"`
-	SubscriptionID string `json:"subscriptionId" yaml:"subscriptionId"`
-	ResourceGroup  string `json:"resourceGroup" yaml:"resourceGroup"`
-	VMType         string `json:"vmType" yaml:"vmType"`
+	providerazure.Config `json:",inline" yaml:",inline"`
```

Comment:
Delegated most of the config back to `cloud-provider-azure`.

However, this, along with other **user-facing** interfaces that got delegated back to `cloud-provider-azure` will be more hideous from the user. For example, if VM type string `"vmss"` changes to `"vmscalesets"` on the newer version of `cloud-provider-azure`, it won't be visible to the user that might not be aware of the usage of `cloud-provider-azure`. 

Well-covered unit testing would definitely help still. But it might even be difficult for the maintainers to keep track of the true size of the exposed interface by the library (e.g., through this `Config` and `cloud-config-file` which will be unmarshalled into this struct).

Or, maybe it is by design that the user is supposed to know the existence of this module and can configure them? If so, is it intended for the maintainer to not having to worry about those interfaces, and treat them as black boxes?

Anyway, in this case (and likely in most cases--in practice), although not ideal, I think it is worth it to delegate it back given that **changes are really rare**, the size of the interface is not too large to have a proper coverage/know what to care about, and there should be a proper deprecation procedures if something changes.

Reply:
"more hideous" -> "more hidden / less visible / less discoverable"? (And I don't think this is a major issue; can be addressed by documentation.)

Reply:
It may also be worth capturing in some way whatever guarantees are intended regarding compatibility of this config with cloud-provider-azure config serialization. Though answer is probably just that this one is a superset?

Reply:
> "more hideous" -> "more hidden / less visible / less discoverable"?

Yes. And also less aware of changes.

> can be addressed by documentation

How would you imagine the documentation to be like? 
Something like A) "we are using cloud-provider-azure, go see their repo for how to use it" or B) "here are the configs" then 'fork' the docs from cloud-provider-azure?
A) would be easier to be maintained, but additional burden to the user. B) would be like a fork to us.

If you ask me I would prefer A). In practice it wouldn't hurt the experience that much given that it seems to always be one-layer.

Reply:
A) makes sense to me, since `cloud-provider-azure` should be keeping their docs up to date (doesn't make sense for us to duplicate the work here).